sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/functions/ScalarFunction.java

```diff
@@ -149,7 +152,10 @@ public interface ScalarFunction<R> extends BoundFunction {
    * @return a result value
    */
   default R produceResult(InternalRow input) {
-    throw new SparkUnsupportedOperationException("_LEGACY_ERROR_TEMP_3146");
+    throw new SparkUnsupportedOperationException(
+            "SCALAR_FUNCTION_NOT_COMPATIBLE",
+            Map.of("scalarFunc", QuotingUtils.quoteIdentifier(name()))
```

Comment:
Could you fix indentation. Should be 2 gaps.

Reply:
> Could you fix indentation. Should be 2 gaps.

Fixed

> LGTM except of a comment. Also update PR's description:
> remove the link to JIRA ticket
> error-classes.json -> error-conditions.json

Fixed